pkg/queue/request_metric.go

```diff
@@ -71,11 +76,18 @@ type appRequestMetricsHandler struct {
 	breaker  *Breaker
 }
 
+type enableTagBasedRoutingKeyType struct{}
+
+var enableTagBasedRoutingKey = enableTagBasedRoutingKeyType{}
+
 // NewRequestMetricsHandler creates an http.Handler that emits request metrics.
 func NewRequestMetricsHandler(next http.Handler,
-	ns, service, config, rev, pod string) (http.Handler, error) {
+	ns, service, config, rev, pod string, enablesTagBasedRouting bool) (http.Handler, error) {
 	keys := append(metrics.CommonRevisionKeys, metrics.PodTagKey,
 		metrics.ContainerTagKey, metrics.ResponseCodeKey, metrics.ResponseCodeClassKey)
+	if enablesTagBasedRouting {
```

Comment:
What's the plan for enabling tag based routing? Is it always optional or will it be turned on and can't be disabled in the future?

Looks like you're introducing two flags, one for adding metric label, one for enabling tag based routing. Can we always add this metric label and use `disabled` as value if tag based routing is disabled? Personally I don't want to expose another knob if this is a feature most people will use.

Reply:
I'm not sure if it will be generally used or not, but in my opinion, it would be better to make the feature enabled as default after some aging periods.

BTW, your suggestion (whenever adding the label) sounds good to me. 
If it becomes not-optional, then I can easily add it to the activator as well. 

If you don't have any other concern about adding the label regardless the feature-enabling flag, I'll reflect your suggestion. Thanks.

Reply:
It should be optional in the beginning, but long term which it should be enabled by default.

Reply:
Then I'd suggest just adding the label without a flag guide.

Reply:
Just for clarity, I would like to check things below.
* The label `tag` is always added by this PR
* If the tag based routing is disabled, then the label `tag` will have a value `disabled`

Is it correct?

Reply:
Right. That's my suggestion.

Reply:
Sounds good. 👍